### ISSUE TITLE
build performous on Windows + IDE integration

### ISSUE THREAD

**Issue by basisbit** [performous/performous#239]
Currently, there is no easy way to build and package performous on Windows. Helpful documentation only exists for cross compiling from Linux for Windows target.

The Visual Studio 2017 IDE (which is free for use) supports handling and building cmake based projects. I tried compiling performous, but various dependencies are missing. These missing libraries seem to be:
> gettext cairo-devel librsvg2-devel libsigc++20-devel glibmm24-devel libxml++-devel boost-devel SDL2-devel libepoxy-devel ffmpeg-devel portaudio-devel help2man redhat-lsb opencv-devel portmidi-devel libjpeg-turbo-devel pango-devel jsoncpp-devel 

I want to try to set up a github dependencies repo where all these libraries (and/or  precompiled libraries + headers if the licenses allow this) will be added as git submodules.
The goal is to be able to easily build performous without having to search days for all the required dependencies and how to compile these.

**Comment by nieknooijens** [comment]
please also create an automated script that will fetch the submodules from that repo and build/configure them correctly (sort of MXE for windows) Performous used to have that for an old visual studio version but it wasn't maintained so due to bitrot we abandoned that script :(

**Comment by basisbit** [comment]
fyi, I stopped any work on this issue because of "dependencies hell". Anyone, please feel free to work on this issue if you want to 😃

**Comment by nieknooijens** [comment]
@basisbit wat might be the best solution is how stepmania handled it.
they have a folder"ext" which contains all the dependencies as submodules, and then they literally scripted everything together using cmake. so if cmake doesn't detect the libs on the system it will compile and link them all.

**Comment by Baklap4** [comment]
As mentioned some time ago.
I've managed to make a build while on Windows 10.
However this succeeded by using the internal Linux Subsystem for Windows.
I've configured visual studio to publish a build to a location the subsystem can reach. then the subsystem created the build, then by hitting f5 (run) it'd start vnc and run it (as a linux binary), far from perfect but it can be done.
I bet it can be tweaked to just crosscompile the binary to a windows binary (while on windows) and test out that i'm not sure about debugging it properly from visual studio though.

However due to the complexity i suggest we close this issue since OP stopped working on this. Meanwhile we should configure semaphore to crosscompile for windows and make a nightly build available see #214 Closing this as most if not all of the devs are working on unix machines (linux or osx)
